src/components/data/FileBrowser.js

```diff
@@ -387,6 +389,34 @@ const BucketBrowser = (({
         noObjectsMessage: Utils.cond(
           [loading, () => 'Loading bucket contents...'],
           [error, () => 'Unable to load bucket contents'],
+          [!!prefix && prefix !== basePrefix && allowEditingFolders, () => div({
+            style: { display: 'flex', flexDirection: 'column', alignItems: 'center' }
+          }, [
+            noObjectsMessage,
+            h(ButtonOutline, {
+              style: { marginTop: '1rem', textTransform: 'none' },
+              onClick: async () => {
+                setBusy(true)
+
+                // Attempt to delete folder placeholder object.
+                // A placeholder object may not exist for the prefix being viewed, so do not an report error for 404 responses.
+                // See https://cloud.google.com/storage/docs/folders for more information on placeholder objects.
+                try {
+                  await Ajax().Buckets.delete(googleProject, bucketName, prefix)
+                } catch (error) {
+                  if (error.status !== 404) {
+                    reportError('Error deleting folder', error)
+                  }
+                }
+
+                setBusy(false)
+
+                // Since prefixes have a trailing slash, the last item in the split array will be an empty string.
+                // Dropping the last two items returns the parent "folder".
+                setPrefix(_.flow(_.split('/'), _.dropRight(2), _.join('/'))(prefix))
+              }
+            }, ['Delete this folder'])
```

Comment:
What does the Delete button up near New Folder do?

Reply:
That button deletes objects selected in the table. We don't currently allow selecting folders in the table because we don't support recursively deleting all objects in a folder.